src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private int threshold;
+    private int capacity;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+        threshold = (int) (capacity * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            grow();
+        }
+        int bucketIndex = getBucketIndex(key);
+        Node<K, V> node = table[bucketIndex];
+        if (node == null) {
+            table[bucketIndex] = new Node<>(key,value,null);
+            size++;
+            return;
+        }
+        while (node != null) {
+            if (Objects.equals(key,node.key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = new Node<>(key,value,null);
+                size++;
+                return;
+            }
+            node = node.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int bucketIndex = getBucketIndex(key);
+        Node<K, V> node = table[bucketIndex];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getBucketIndex(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % table.length;
+    }
+
+    private void grow() {
+        size = 0;
+        Node<K, V>[] oldNodes = table;
+        table = new Node[capacity * 2];
+        threshold = (int) (capacity * LOAD_FACTOR);
```

Comment:
```suggestion
        Node<K, V>[] oldNodes = table;
        capacity *= 2;
        table = new Node[capacity];
        threshold = (int) (capacity * LOAD_FACTOR);
```
you are not updating capacity in this method